### ISSUE TITLE
The memory usage is too high; the computer’s memory is struggling to keep up.

### ISSUE THREAD

**Issue by voidfeng** [johnsoncodehk/volar#4653]
### Vue - Official extension or vue-tsc version

v2.0.28

### VSCode version

1.92.0

### Vue version

3.4.33

### TypeScript version

5.4.0

### System Info

```shell
System:
    OS: macOS 14.5
    CPU: (8) arm64 Apple M1
    Memory: 97.19 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.14.0 - ~/.nvm/versions/node/v20.14.0/bin/node
    npm: 10.7.0 - ~/.nvm/versions/node/v20.14.0/bin/npm
    bun: 1.1.12 - ~/.bun/bin/bun
  Browsers:
    Brave Browser: 124.1.65.130
    Chrome: 127.0.6533.89
    Chrome Canary: 129.0.6639.0
    Edge: 127.0.2651.74
    Safari: 17.5
```

### Steps to reproduce

Please watch the video.

### What is expected?

Reduce memory usage.

### What is actually happening?

The memory usage is extremely high.

### Any additional comments?

https://github.com/user-attachments/assets/ecaaba9c-faee-45ac-bf5a-ee9314b63f35

**Comment by davidmatter** [comment]
Please provide more information about your project and your project config. Generally, large Vue/TS projects require a lot of memory by default as type-checking is quite resource intensive.

**Comment by coder-virjay** [comment]
> Please provide more information about your project and your project config. Generally, large Vue/TS projects require a lot of memory by default as type-checking is quite resource intensive. I tried deleting some Vue and TS files, and the memory usage decreased accordingly. When I completely deleted the src folder, the memory usage was very low.

I’ve found that as the number of Vue and TS files increases, the memory usage also rises. My project has over six hundred Vue and TS files. The file below is the project’s dependencies.

[package.json](https://github.com/user-attachments/files/16517276/package.json)

**Comment by QFifteen** [comment]
I have the same problem. The environment is consistent. It's a small project.

**Comment by casesolved-co-uk** [comment]
Yes resource requirements are horrific, especially when using remote on Remote - SSH in vscode. In the end I've had to remove the extension just on the server because the extension would either crash or take my server down.
Is there any way to run this locally even when editing over SSH?

**Comment by johnsoncodehk** [comment]
There are many factors that affect TS's memory usage, and without reproduction, we will not be able to investigate the problem.
